### ISSUE TITLE
The difference between the code and the official implementation of mean teacher

### ISSUE THREAD

**Issue by tea321000** [HiLab-git/SSL4MIS#17]
Hi  Xiangde,

I checked the official implementation of mean teacher and found that consistency loss is calculated on[ the samples in the entire batch](https://github.com/CuriousAI/mean-teacher/blob/master/pytorch/main.py#L263), not just [unlabeled samples](https://github.com/HiLab-git/SSL4MIS/blob/master/code/train_mean_teacher_3D.py#L157). 

`cons_logit` is assigned from `logit1` or `logit2` according to different settings of [`args.logit_distance_cost`](https://github.com/CuriousAI/mean-teacher/blob/master/pytorch/main.py#L247-L251), and they are calculated through different [fully connected layers](https://github.com/CuriousAI/mean-teacher/blob/master/pytorch/mean_teacher/architectures.py#L168) to loosen the link between the classification prediction and the consistency cost. 

I was wondering if my understanding of the code is biased, or if there is a bug in the implementation.

Sincerely,
tea

**Comment by Luoxd1996** [comment]
Hi, 
It's just my implementation, but in my experiments, I found that our implementation can achieve better results, you can implement or adapt the official implementation for your own tasks.
Best,
Xiangde.

**Comment by tea321000** [comment]
Got it. Thanks for your quick reply.
